### PR TITLE
No team role/rank if round over

### PR DESCRIPTION
**Pull request by Natty-Boh**
But will still update nickname as normal

...probably. i didnt test it 👀

### CODE REVIEW

File: src/service/nicknameService.ts [hunk 1]
```diff
@@ -60,19 +62,20 @@ export function buildUserRank(member: GuildMember): UserRank {
...
-        await processMember(member, user, userRank.id);
+        await processMember(member, user, userRank.id, isRoundOver);
     }
 }
 
-function buildRankNickname(user: UserData) {
-    const rank = rankMap.get(user.rank) ?? "";
+function buildRankNickname(user: UserData, isRoundOver: boolean) {
+    const rank = isRoundOver ? "" : rankMap.get(user.rank) ?? "";
     return `${rank} ${user.name}`;
```
Comment (TotallyNot) [inline src/service/nicknameService.ts:79]: this will result in a leading whitespace when `isRoundOver` is true

File: src/service/util/team.ts [hunk 2]
```diff
@@ -26,3 +27,13 @@ export function teamFromControl(control: number): number {
...
 }
+
+export function isRoundOver(world: CountryData[]): boolean {
+    const axisSpanws = world.filter(
+        (c) => c.isActiveSpawn && c.isSpawn && c.initialControl === 0
+    );
+    const alliesSpanws = world.filter(
+        (c) => c.isActiveSpawn && c.isSpawn && c.initialControl === 100
+    );
+    return alliesSpanws.length === 0 || axisSpanws.length === 0;
```
Comment (TotallyNot) [inline src/service/util/team.ts:38]: nitpicky, but `some` would be better suited for this than `filter` + `length`

File: src/service/nicknameService.ts (hunk 1 above)
Comment (Natty-Boh) [reply src/service/nicknameService.ts:79]: no discord trims it, this is already what happens when user is not in round

File: src/service/util/team.ts (hunk 2 above)
Comment (Natty-Boh) [reply src/service/util/team.ts]: changed

File: src/service/util/team.ts [hunk 3]
```diff
@@ -26,3 +27,13 @@ export function teamFromControl(control: number): number {
...
 }
+
+export function isRoundOver(world: CountryData[]): boolean {
+    const axisSpawns = world.some(
+        (c) => c.isActiveSpawn && c.isSpawn && c.initialControl === 0
+    );
+    const alliesSpawns = world.some(
+        (c) => c.isActiveSpawn && c.isSpawn && c.initialControl === 100
+    );
+    return alliesSpawns || axisSpawns;
```
Comment (TotallyNot) [inline src/service/util/team.ts:38]: yes, but now it needs to be inverted >.>
`!alliesSpawns || !axisSpawn`

Comment (TotallyNot) [reply src/service/util/team.ts:38]: actually, why check for both `isSpawn` and `isActiveSpawn`? Isn't that redundant?

**Review by TotallyNot — APPROVED** [review]
lgtm

**Pull request merged**